src/ui/menu_nbgl.c

```diff
@@ -34,91 +34,81 @@ void app_quit(void) {
     os_sched_exit(-1);
 }
 
-// home page definition
-void ui_menu_main(void) {
-// This parameter shall be set to false if the settings page contains only information
-// about the application (version , developer name, ...). It shall be set to
-// true if the settings page also contains user configurable parameters related to the
-// operation of the application.
-#define SETTINGS_BUTTON_ENABLED (true)
-
-    nbgl_useCaseHome(APPNAME,
-                     &C_app_boilerplate_64px,
-                     NULL,
-                     SETTINGS_BUTTON_ENABLED,
-                     ui_menu_settings,
-                     app_quit);
-}
-
 //  -----------------------------------------------------------
 //  --------------------- SETTINGS MENU -----------------------
 //  -----------------------------------------------------------
-
-static const char* const INFO_TYPES[] = {"Version", "Developer"};
-static const char* const INFO_CONTENTS[] = {APPVERSION, "Ledger"};
+#define SETTING_INFO_NB 2
+static const char* const INFO_TYPES[SETTING_INFO_NB] = {"Version", "Developer"};
+static const char* const INFO_CONTENTS[SETTING_INFO_NB] = {APPVERSION, "Ledger"};
 
 // settings switches definitions
 enum { DUMMY_SWITCH_1_TOKEN = FIRST_USER_TOKEN, DUMMY_SWITCH_2_TOKEN };
 enum { DUMMY_SWITCH_1_ID = 0, DUMMY_SWITCH_2_ID, SETTINGS_SWITCHES_NB };
 
-static nbgl_layoutSwitch_t switches[SETTINGS_SWITCHES_NB] = {0};
+static nbgl_contentSwitch_t switches[SETTINGS_SWITCHES_NB] = {0};
 
-static bool nav_callback(uint8_t page, nbgl_pageContent_t* content) {
-    UNUSED(page);
+static const nbgl_contentInfoList_t infoList = {
+    .nbInfos = SETTING_INFO_NB,
+    .infoTypes = INFO_TYPES,
+    .infoContents = INFO_CONTENTS,
```

Comment:
Would it be feasible to reverse this order? a list of (type, content) values
Defining the two values separetely is weird

Reply:
Probably no as it would break everything

Reply:
Could be another content type, but i don't think it's worth it...